python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -259,21 +332,331 @@ def scaleModel(self, model, star, inPlace=True, nb90Rots=0):
+
+        Parameters
+        ----------
+        annulusImage : `~lsst.afw.image.MaskedImageF`
+            An image containing an annulus of the given model.
+
+        Returns
+        -------
+        annularFlux: float
+            The annular flux of the PSF model at the radius where the flux of the given star is determined.
+        """
+        andMask = reduce(
+            ior, (annulusImage.mask.getPlaneBitMask(bm) for bm in self.warper.config.badMaskPlanes)
+        )
+        self.missedStatsControl.setAndMask(andMask)
+        annulusStat = makeStatistics(annulusImage, self.missedStatsFlag, self.missedStatsControl)
+        return annulusStat.getValue()
+
+    def findPsfAnnularFlux(self, brightStarStamp, maskedModel):
+        """Find the annular flux of the PSF model at the radius where the flux of the given star is
+        determined.
+
+        Parameters
+        ----------
+        brightStarStamp : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamp`
+            A stamp of a bright star to be subtracted.
+        maskedModel : `~lsst.afw.image.MaskedImageF`
+            A masked image of the PSF model.
+
+        Returns
+        -------
+        annularFlux: float
+            The annular flux of the PSF model at the radius where the flux of the given star is determined.
+        """
+        annulusImage = self.initAnnulusImage()
+        annulus = self.createAnnulus(brightStarStamp)
+        annulus.copyMaskedImage(maskedModel, annulusImage)
+        annularFlux = self.applyStatsControl(annulusImage)
+        return annularFlux
+
+    def findPsfAnnularFluxes(self, brightStarStamps):
+        """Find the annular fluxes of the given PSF model.
+
+        Parameters
+        ----------
+        brightStarStamps : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamps`
+            The stamps of stars that will be subtracted from the exposure.
+
+        Returns
+        -------
+        PsfAnnularFluxes: numpy.array
+            A two column numpy.array containing annular fluxes of the PSF at radii where the flux for stars
+            exist (could be found).
+
+        Notes
+        -----
+        While the PSF model is normalized at a certain radius, the flux of a star at that radius might be
+        impossible to find. Therefore, we have to scale the PSF model considering a radius where the star has
+        an identified flux. To do that, the flux of the model should be found and used to adjust the scaling
+        step.
+        """
+        outerRadii = []
+        annularFluxes = []
+        maskedModel = MaskedImageF(self.model.image)
+        # the model has wrong bbox values. Should be fixed in extended_psf.py?
+        maskedModel.setXY0(0, 0)
+        for star in brightStarStamps:
+            if star.optimalOuterRadius not in outerRadii:
+                annularFlux = self.findPsfAnnularFlux(star, maskedModel)
+                outerRadii.append(star.optimalOuterRadius)
```

Comment:
Here, the flux of the PSF model at the radius that the star is normalised is found and stored. This value is used later to scale the model to the brightness level of the star.

In general, the first inner radius is the one that is used to normalise stars stamps in the ProcessBrightStarsTask. However, some stars have no flux for that. Therefore, we have to use another radius to find a flux and scale the model based on that value. The code starts looking for an existing flux for a star like that by taking the old outer radius as the new inner radius, and computes the new outer radius. This process continues until the code finds a flux within an annulus.

If the code find no flux for the star, it skips that star and do not subtract it from the exposure. 
For each exposure, stars that are not subtracted will be stored in a fits file. 
In future, we will deal with these type of stars and add some methods to subtract them in a proper way.

Reply:
Ok, so it keeps looking outward if it can't find suitable aperture fluxes (because of masked/bad pixels?). I think I need more explanation about the choice of radius for normalization but it shouldn't go here.